doc/fileformat.rst

```diff
@@ -0,0 +1,74 @@
+OSI File Format
+----------------
+
+To save multiple serialized OSI messages into one trace file we use the length of each OSI message and save it before the actual OSI message. 
+The length is represented by the first four bytes which are a little endian unsigned int that represents the length of the followed message, not including the integer itself.
+If you happen to have a trace file which uses ``$$__$$`` separation you can convert it to the official OSI trace file by running ``python3 txt2osi.py -d trace.txt``.
+
+To read to content of a serialized txt/osi trace file we also provide a converter ``osi2read.py``.
+See the usage below:
+
+.. code-block:: bash
+
+    python3 osi2read.py -d trace.osi -o readable_trace
+    python3 osi2read.py -d trace.txt -f separated -o readable_trace
+
+which outputs a ``readable_trace.txth`` which can be opened by any text editor.
+
+If you want to generate a valid OSI trace file which can be used as an input for the `osi-validator <https://github.com/OpenSimulationInterface/osi-validation>`_ or the `osi-visualizer <https://github.com/OpenSimulationInterface/osi-visualizer>`_ see the example script in python below:
+
+.. code-block:: python
+
+    from osi3.osi_sensorview_pb2 import SensorView
+    import struct
+
+    def main():
+        """Initialize SensorView"""
+        f = open("test_trace.osi", "ab")
+        sensorview = SensorView()
+
+        sv_ground_truth = sensorview.global_ground_truth
+        sv_ground_truth.version.version_major = 3
+        sv_ground_truth.version.version_minor = 0
+        sv_ground_truth.version.version_patch = 0
+
+        sv_ground_truth.timestamp.seconds = 4
+        sv_ground_truth.timestamp.nanos = 54999999
+
+        stationary_object = sv_ground_truth.stationary_object.add()
+        stationary_object.id.value = 114
+
+        for i in range(11):
```

Comment:
I do not understand the loop here ... Don't you have to increment the time as well? And the <L does not stand for a Long type?

Reply:
Yes the time increment should be also in the loop, will add it. The '<L' stands for little endian long type which in python is defined as a 32bit int (see pmai's[ comment](https://github.com/OpenSimulationInterface/osi-visualizer/issues/30#issuecomment-557580286)). Since OSMP are restricted to 32bit I thought we would use than the 32bit int.

Reply:
He says "LE"?

Reply:
LE: **L**ittle **E**ndian int 32bit